### PR TITLE
Dump CUDA source on compilation error

### PR DESCRIPTION
**Pull request by niboshi**
Implements #190

### CODE REVIEW

**Comment by okuta** [comment]
Thank you for creating this PR.
How about merge source code to CompileException?

**Comment by niboshi** [comment]
Do you mean like this?

**Comment by unnonouno** [comment]
Do you need to write a test?

**Comment by niboshi** [comment]
As I told you internally, IMO unit test is not necessary because this feature is only for the case where catastrophic failure has already occurred.

File: cupy/cuda/compiler.py [hunk 1]
```diff
@@ -54,13 +56,28 @@ def compile_using_nvrtc(source, options=(), arch=None):
             cu_file.write(source)
 
         prog = _NVRTCProgram(source, cu_path)
-        ptx = prog.compile(options)
+        try:
+            ptx = prog.compile(options)
+        except CompileException as e:
+            dump = bool(int(
```
Comment (unnonouno) [inline cupy/cuda/compiler.py]: Do we need to check cast error?

**Comment by niboshi** [comment]
Fixed.

File: docs/source/reference/environment.rst [hunk 2]
```diff
@@ -4,11 +4,15 @@ Environment variables
...
-|                    | See :ref:`overview` for details.                   |
-+--------------------+----------------------------------------------------+
++------------------------------------+----------------------------------------------------+
+| ``CUPY_CACHE_DIR``                 | Path to the directory to store kernel cache.       |
+|                                    | ``$(HOME)/.cupy.kernel_cache`` is used by default. |
+|                                    | See :ref:`overview` for details.                   |
++------------------------------------+----------------------------------------------------+
+| ``CUPY_DUMP_CUDA_SOURCE_ON_ERROR`` | If set to 1, when CUDA kernel compilation fails,   |
+|                                    | CuPy dumps CUDA kernel code to standard error.     |
+|                                    | It's disabled by default.                          |
```
Comment (unnonouno) [inline docs/source/reference/environment.rst]: `It is` is better in the document.

**Comment by niboshi** [comment]
Fixed!

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**